src/PlatformCompatibilityTests/ServiceControlCompatibilityTests/SuccessfulRetryTests.cs

```diff
@@ -0,0 +1,74 @@
+﻿namespace ServiceControlCompatibilityTests
+{
+    using Autofac;
+    using NServiceBus;
+    using NUnit.Framework;
+    using System;
+    using System.Threading.Tasks;
+
+    [TestFixture]
+    class SuccessfulRetryTests : SqlScTest
+    {
+        [TestCaseSource(nameof(AllTransports))]
+        public async Task Can_successfully_retry_a_failed_message(Type transportDetailType)
+        {
+            var transportDetails = StartUp(transportDetailType);
+            await RunTest(transportDetails);
+        }
+
+        async Task RunTest(ITransportDetails transportDetails)
+        {
+            var handlerInstance = new TestMessageHandler();
+            var builder = new ContainerBuilder();
+            builder.RegisterInstance(handlerInstance);
+            var container = builder.Build();
```

Comment:
IMO this whole block about registering the handler instance seem to be unrelated to the test. Could it be moved out to somewhere else? Some sort of extension/helper or somehow done in the Endpoint creation.